cli/crawl-cli.js

```diff
@@ -34,8 +35,65 @@ program
     .option('--config <path>', 'crawl configuration file')
     .option('--autoconsent-action <action>', 'dismiss cookie popups. Possible values: optout, optin')
     .option('--chromium-version <version_number>', 'use custom version of chromium')
+    .option('--selenium-hub <url>', 'selenium hub endpoint to request browsers from')
     .parse(process.argv);
 
+/**
+ * @param {string} outputPath
+ * @param {URL} url
+ * @param {string} fileType file extension, defaults to 'json'
+ */
+function createOutputPath(outputPath, url, fileType = 'json') {
+    return path.join(outputPath, `${createUniqueUrlName(url)}.${fileType}`);
+}
+
+/**
+ * @param {Array<string|{url:string, dataCollectors?:BaseCollector[]}>} inputUrls
+ * @param {function} logFunction
+ * @param {string} outputPath
+ */
+function filterUrls(inputUrls, logFunction, outputPath) {
+    return new Promise((resolveFilterUrls, rejectFilterUrls) => {
+        asyncLib.filter(inputUrls, (item, filterCallback) => {
```

Comment:
is performance here worth the the complexity of doing filtering async?

Reply:
I didn't do a detailed profiling, but with hundreds of thousands URLs, this takes quite a lot of time (tens of seconds). Depending on the FS speed it could make a difference I think.
The latest async lib supports promises, so if/when we upgrade it it should look less ugly.